libs/langchain-google-cloud-sql-pg/src/engine.ts

```diff
@@ -96,12 +134,125 @@ class PostgresEngine {
     return new PostgresEngine(PostgresEngine._createKey, engine)
   }
 
+  /**
+   * Create a PostgresEngine instance from an Knex instance.
+   * 
+   * @param engine knex instance
+   * @returns PostgresEngine instance from a knex instance
+   */
+  static async from_engine(engine: knex.Knex<any, any[]>) {
+    return new PostgresEngine(PostgresEngine._createKey, engine)
+  }
+
+  /**
+   * Create a PostgresEngine instance from arguments.
+   * 
+   * @param url URL use to connect to a database
+   * @returns PostgresEngine instance
+   */
+  static async from_engine_args(url: string | knex.Knex.StaticConnectionConfig) {
+
+    const driver = 'postgresql+asyncpg';
+
+    // if((typeof url === "string") && !url.startsWith(driver) || (url instanceof URL && url.protocol !== driver)){
+    if((typeof url === "string") && !url.startsWith(driver)){
+      throw "Driver must be type 'postgresql+asyncpg'"
+    }
+
+    const dbConfig: knex.Knex.Config<any> = {
```

Comment:
Can we also support the user passing in a map of configs to extend the dbConfig object?

Reply:
Could you be more specific? The dbConfig here uses the URL to connect to the database, and this already contains the database user, I don't think it's necessary to add the user to the configuration.